frontend/public/actions/k8s.ts

```diff
@@ -292,47 +291,39 @@ export const watchK8sList = (
   pollAndWatch();
 };
 
-export const setAPIGroups = (value: number) => action(ActionType.SetAPIGroups, { value });
-
-export const watchAPIServices = () => (dispatch, getState) => {
-  if (getState().k8s.has('apiservices') || POLLs[apiGroups]) {
-    return;
-  }
-  dispatch({ type: ActionType.GetResourcesInFlight });
-
-  k8sList(APIServiceModel, {})
-    .then(() =>
+export const startAPIDiscovery = () => (dispatch) => {
+  const reduxID = makeReduxID(CustomResourceDefinitionModel, {});
+  checkAccess({
+    group: CustomResourceDefinitionModel.apiGroup,
+    resource: CustomResourceDefinitionModel.plural,
+    verb: 'list',
+  }).then((res) => {
+    if (res.status.allowed) {
+      // eslint-disable-next-line no-console
+      console.log('API discovery method: Watching');
+      // Watch CRDs and dispatch refreshAPI action whenever an event is received
       dispatch(
         watchK8sList(
-          makeReduxID(APIServiceModel, {}),
+          reduxID,
           {},
-          APIServiceModel,
-          (id: string, events: K8sEvent[]) => {
-            // Only re-run API discovery on added or removed API services. A
-            // misbehaving API service can trigger frequent watch updates,
-            // which could cause console to thrash.
-            return events.some(({ type }) => type !== 'MODIFIED') ? getResources() : _.noop;
-          },
+          CustomResourceDefinitionModel,
+          // Only re-run API discovery on added or removed CRDs.
+          (_id: string, events: K8sEvent[]) =>
+            events.some((e) => e.type !== 'MODIFIED') ? getResources() : _.noop,
         ),
-      ),
-    )
-    .catch(() => {
-      const poller = () =>
-        consoleFetchJSON('api/kubernetes/apis')
-          .then((d) => {
-            if (d.groups.length !== getState().k8s.getIn(['RESOURCES', apiGroups], 0)) {
-              dispatch(getResources());
-            }
-            dispatch(setAPIGroups(d.groups.length));
-          })
-          .catch((e) => {
-            // eslint-disable-next-line no-console
-            console.warn('Could not fetch API groups', e);
-          });
-
-      POLLs[apiGroups] = setInterval(poller, 30 * 1000);
-      poller();
-    });
+      );
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('API discovery method: Polling');
+      // Poll API discovery every 30 seconds since we can't watch CRDs
+      dispatch(getResources());
+      if (POLLs[apiDiscovery]) {
+        clearTimeout(POLLs[apiDiscovery]);
+        delete POLLs[apiDiscovery];
+      }
+      POLLs[apiDiscovery] = setTimeout(() => dispatch(startAPIDiscovery()), 30000);
```

Comment:
I feel like every 30s might be a little too aggressive, particularly now that we re-run discovery on the operand pages... We might pull this back to every minute or every few minutes. (If there are other common scenarios where we need to re-run discovery more often, we can target those like we do for operands.)